htlcswitch/hop/error_encryptor.go

```diff
@@ -145,15 +178,39 @@ func (s *SphinxErrorEncrypter) IntermediateEncrypt(
 
 // Type returns the identifier for a sphinx error encrypter.
 func (s *SphinxErrorEncrypter) Type() EncrypterType {
-	return EncrypterTypeSphinx
+	// If no blinding point is present, then this is a regular sphinx
+	// encrypter.
+	if s.BlindingKey.blindingPoint == nil {
+		return EncrypterTypeSphinx
+	}
+
+	// When we have a non-nil blinding point, the encrypter type depends
+	// on the blinding key.
+	if s.BlindingKey.isIntroduction {
+		return EncrypterTypeIntroduction
+	}
+
+	return EncrypterTypeBlindedRelay
 }
 
 // Encode serializes the error encrypter' ephemeral public key to the provided
 // io.Writer.
 func (s *SphinxErrorEncrypter) Encode(w io.Writer) error {
 	ephemeral := s.EphemeralKey.SerializeCompressed()
-	_, err := w.Write(ephemeral)
-	return err
+	if _, err := w.Write(ephemeral); err != nil {
+		return err
+	}
+
+	if s.Type() == EncrypterTypeSphinx {
+		return nil
+	}
+
+	blinding := s.BlindingKey.blindingPoint.SerializeCompressed()
```

Comment:
If we are not a normal sphinx hop, then we will persist the route blinding point to disk!